src/scripts/main.js

```diff
@@ -1,3 +1,241 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const body = document.querySelector('body');
+const listThead = [...table.children[0].children[0].children];
+
+table.addEventListener('click', (e) => {
+  const listElements = [...table.children[1].children];
+
+  if (!e.target.parentElement.classList.contains('active')) {
+    listElements.forEach(tr => tr.classList.remove('active'));
+  }
+
+  if (e.target.tagName === 'TD') {
+    e.target.parentElement.classList.add('active');
+  }
+
+  if (e.target.tagName === 'TH') {
+    let sortElements = '';
+    const index = e.target.cellIndex;
+    const tableCell = e.target.textContent;
+
+    if (!e.target.classList.contains(tableCell)) {
+      listThead.forEach(th => {
+        return th.classList
+          .remove('asc', 'Name', 'Position', 'Office', 'Age', 'Salary');
+      });
+    }
+
+    switch (tableCell) {
+      case 'Name':
+      case 'Position':
+      case 'Office':
+        e.target.classList.toggle('asc');
+
+        if (e.target.classList.contains('asc')) {
+          sortElements = listElements.sort((a, b) => {
+            return a.children[index].textContent
+              .localeCompare(b.children[index].textContent);
+          });
+        } else {
+          sortElements = listElements.sort((a, b) => {
+            return b.children[index].textContent
+              .localeCompare(a.children[index].textContent);
+          });
+        }
+        e.target.classList.add(tableCell);
```

Comment:
You can make your code shorter and easier to read if you sort your array in desc first, and than just reverse it if you need asc order
```suggestion
       sortElements = listElements.sort((a, b) => {
            return b.children[index].textContent
              .localeCompare(a.children[index].textContent
        });
        
        if (e.target.classList.contains('asc')) {
          sortElements.reverse();
        };
        
        e.target.classList.add(tableCell);
```